orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandlerTest.kt

```diff
@@ -1028,6 +1029,57 @@ object CompleteStageHandlerTest : SubjectSpek<CompleteStageHandler>({
         verify(queue).push(StartStage(pipeline.stageByRef("2=1>1")))
       }
     }
+
+    setOf(TERMINAL, FAILED_CONTINUE).forEach { taskStatus ->
+      describe("when a stage's task fails with $taskStatus status and should assign the taskExceptionDetails to stageContext exception") {
+        val pipeline = pipeline {
+          stage {
+            refId = "1"
+            type = multiTaskStage.type
+            multiTaskStage.plan(this)
+            tasks[0].status = taskStatus
+            tasks[0].taskExceptionDetails["exception"] = mapOf("timestamp" to "", "exceptionType" to "", "operation" to "", "details" to
+              mapOf("responseBody" to "", "kind" to "", "error" to "", "errors" to arrayListOf("something"), "url" to "", "rootException" to "",
+                "status" to ""), "shouldRetry" to false)
+            status = RUNNING
+          }
+        }
+        val message = CompleteStage(pipeline.stageByRef("1"))
+
+        beforeGroup {
+          whenever(repository.retrieve(PIPELINE, message.executionId)) doReturn pipeline
+        }
+
+        afterGroup(::resetMocks)
+
+        on("receiving the message") {
+          subject.handle(message)
+        }
+
+        it("updates the stage context to store the task exception") {
+          verify(repository).storeStage(
+            check {
+              assertThat(it.status).isEqualTo(taskStatus)
+              assertThat(((((it.context["exception"] as Map<*, *>)["details"]) as Map<*, *>)["errors"] as ArrayList<*>)[2]).isEqualTo((((it.tasks[0].taskExceptionDetails["exception"] as Map<*, *>)["details"] as Map<*, *>)["errors"] as ArrayList<*>)[0])
```

Comment:
Can you split this over a couple of lines? It's very difficult to understand.

Reply:
Had to remove this test case and add a new one